libraries/rush-lib/src/api/Subspace.ts

```diff
@@ -321,4 +325,71 @@ export class Subspace {
   public _addProject(project: RushConfigurationProject): void {
     this._projects.push(project);
   }
+
+  /**
+   * Returns hash value of injected dependencies in related package.json.
+   * @beta
+   */
+  public getPackageJsonInjectedDependenciesHash(): string | undefined {
+    const allPackageJson: IPackageJson[] = [];
+
+    const relatedProjects: RushConfigurationProject[] = [];
+    const subspacePnpmfileShimSettings: ISubspacePnpmfileShimSettings =
+      SubspacePnpmfileConfiguration.getSubspacePnpmfileShimSettings(this._rushConfiguration, this);
+
+    for (const rushProject of this.getProjects()) {
+      const injectedDependencies: Array<string> =
+        subspacePnpmfileShimSettings?.subspaceProjects[rushProject.packageName]?.injectedDependencies || [];
+      if (injectedDependencies.length === 0) {
+        continue;
+      }
+
+      const injectedDependencySet: Set<string> = new Set(injectedDependencies);
+
+      for (const dependencyProject of rushProject.dependencyProjects) {
+        if (injectedDependencySet.has(dependencyProject.packageName)) {
+          relatedProjects.push(dependencyProject);
+        }
+      }
+    }
+
+    // this means no injected dependencies found for current subspace
+    if (relatedProjects.length === 0) {
+      return undefined;
+    }
+
+    // get all related package.json
+    while (relatedProjects.length > 0) {
+      const rushProject: RushConfigurationProject = relatedProjects.pop()!;
+      allPackageJson.push(this._getImportantFieldsInPackageJson(rushProject.packageJson));
+      relatedProjects.push(...rushProject.dependencyProjects);
+    }
+
+    allPackageJson.sort((pa, pb) => pa.name.localeCompare(pb.name));
+    const hash: crypto.Hash = crypto.createHash('sha1');
+    for (const packageFile of allPackageJson) {
+      hash.update(JSON.stringify(packageFile));
+    }
+
+    const packageJsonInjectedDependenciesHash: string = hash.digest('hex');
+
+    return packageJsonInjectedDependenciesHash;
+  }
+
+  private _getImportantFieldsInPackageJson(packageJson: IPackageJson): IPackageJson {
+    // collect fields that could update the `pnpm-lock.yaml`
+    const packageJsonWithImportantFields: IPackageJson = {
+      name: packageJson.name,
+      version: packageJson.version,
+      bin: packageJson?.bin,
+      dependencies: packageJson?.dependencies,
+      devDependencies: packageJson?.devDependencies,
+      peerDependencies: packageJson?.peerDependencies,
+      optionalDependencies: packageJson?.optionalDependencies,
+      dependenciesMeta: packageJson?.dependenciesMeta,
+      peerDependenciesMeta: packageJson?.peerDependenciesMeta,
+      resolutions: packageJson?.resolutions
+    };
+    return packageJsonWithImportantFields;
+  }
```

Comment:
```suggestion
  private _getImportantFieldsInPackageJson({
    name,
    version,
    bin,
    dependencies,
    devDependencies,
    peerDependencies,
    optionalDependencies,
    dependenciesMeta,
    peerDependenciesMeta,
    resolution,
  }: IPackageJson): IPackageJson {
    // collect fields that could update the `pnpm-lock.yaml`
    const packageJsonWithImportantFields: IPackageJson = {
      name,
      version,
      bin,
      dependencies,
      devDependencies,
      peerDependencies,
      optionalDependencies,
      dependenciesMeta,
      peerDependenciesMeta,
      resolution,
    };
    return packageJsonWithImportantFields;
  }
```

This avoids a bunch of `packageJson` references.

Also, given that this function doesn't refer to `this`, it would be more efficiently bundled as a loose function instead of a class member.

You could also just inline this in `getPackageJsonInjectedDependenciesHash`.